### PR TITLE
LOCK and collection creation revision

### PR DESCRIPTION
**Pull request by dizzzz**
Revise LOCK strategy when copying a collection
Revise collection creation : always update metadata when creating new collection

### CODE REVIEW

File: java/src/org/exist/jms/replication/subscribe/ReplicationJmsListener.java [hunk 1]
```diff
@@ -729,9 +720,9 @@ private Collection getOrCreateCollection(final XmldbURI sourcePath, final Option
             throw new MessageReceiveException(t.getMessage(), t);
 
         } finally {
-            releaseLock(newCollection, Lock.LockMode.WRITE_LOCK);
+            releaseLock(collection, Lock.LockMode.WRITE_LOCK);
```
Comment (shabanovd) [inline java/src/org/exist/jms/replication/subscribe/ReplicationJmsListener.java:723]: where did you allocate this lock?

Comment (dizzzz) [reply java/src/org/exist/jms/replication/subscribe/ReplicationJmsListener.java:723]: I hoped org.exist.storage.DBBroker#getOrCreateCollection or org/exist/storage/NativeBroker.java:778 .....

Comment (dizzzz) [reply java/src/org/exist/jms/replication/subscribe/ReplicationJmsListener.java:723]: I have been struggling with this subject for many years, it got never clear to me how things worked. Lack of documentation :-) Do you think I am wrong here?

Comment (adamretter) [reply java/src/org/exist/jms/replication/subscribe/ReplicationJmsListener.java:723]: @dizzzz Yeah the documentation sucks, you really have to dig deep into the code to understand. I am currently replacing the locking implementation in eXist with something more consistent.

What you need to know most likely right now is that:
1. `NativeBroker#openCollection` returns a locked Collection, you must manually release that lock when you are finished with the Collection.
2. `NativeBroker#getCollection` returns an unlocked Collection.
3. `NativeBroker#getOrCreateCollection` returns a unlocked Collection or locked Collection(s) depending on whether get or create occurs; creation will write lock all the collections in the path that it needs to create. You DO NOT need to manually release these locks as ownership is passed to the Transaction which will auto-free them.

**Pull request merged**